From f3ee8452f244a748f48419c046618934b000d3f3 Mon Sep 17 00:00:00 2001
From: Ping Gao <ping.a.gao@intel.com>
Date: Wed, 6 Sep 2017 19:57:57 +0800
Subject: [PATCH 0927/1132] drm/i915/gvt: Add a module parameter to disable
 conformance check

By default the guest conformance check is enabled.

Signed-off-by: Ping Gao <ping.a.gao@intel.com>
Reviewed-by: Kevin Tian
Reviewed-by: Singh, Satyeshwar <satyeshwar.singh@intel.com>

Conflicts:
	drivers/gpu/drm/i915/i915_params.c
	drivers/gpu/drm/i915/i915_params.h

Reviewed-on:
Reviewed-by: He, Min <min.he@intel.com>
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>

V2: rebase 4.19
Signed-off-by: Xinyun Liu <xinyun.liu@intel.com>
---
 drivers/gpu/drm/i915/gvt/scheduler.c | 3 ++-
 drivers/gpu/drm/i915/i915_params.c   | 3 +++
 drivers/gpu/drm/i915/i915_params.h   | 3 ++-
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/scheduler.c b/drivers/gpu/drm/i915/gvt/scheduler.c
index 6dde102..c78ed32 100644
--- a/drivers/gpu/drm/i915/gvt/scheduler.c
+++ b/drivers/gpu/drm/i915/gvt/scheduler.c
@@ -712,7 +712,8 @@ static int dispatch_workload(struct intel_vgpu_workload *workload)
 
 	ret = intel_gvt_scan_and_shadow_workload(workload);
 
-	if (intel_gvt_vgpu_conformance_check(vgpu, ring_id))
+	if (i915_modparams.enable_conformance_check
+			&& intel_gvt_vgpu_conformance_check(vgpu, ring_id))
 		gvt_err("vgpu%d unconformance guest detected\n", vgpu->id);
 
 	if (ret)
diff --git a/drivers/gpu/drm/i915/i915_params.c b/drivers/gpu/drm/i915/i915_params.c
index 4ffdd533..5c6e7dc 100644
--- a/drivers/gpu/drm/i915/i915_params.c
+++ b/drivers/gpu/drm/i915/i915_params.c
@@ -251,6 +251,9 @@ i915_param_named_unsafe(domain_plane_owners, ullong, 0400,
 	planes owner: 3C:2 2C:2 1C:1 4B:1 3B:1 2B:1 1B:0 4A:0 3A:0 2A:1 1A:0 \
 	(0x0 - default value)");
 
+i915_param_named(enable_conformance_check, bool, 0400,
+	"To toggle the GVT guest conformance feature(default:true)");
+
 static __always_inline void _print_param(struct drm_printer *p,
 					 const char *name,
 					 const char *type,
diff --git a/drivers/gpu/drm/i915/i915_params.h b/drivers/gpu/drm/i915/i915_params.h
index 6b0f98c..d902bed 100644
--- a/drivers/gpu/drm/i915/i915_params.h
+++ b/drivers/gpu/drm/i915/i915_params.h
@@ -75,7 +75,8 @@ struct drm_printer;
 			PVMMIO_ELSP_SUBMIT | PVMMIO_PLANE_UPDATE \
 			| PVMMIO_PLANE_WM_UPDATE | PVMMIO_PPGTT_UPDATE \
 			| PVMMIO_GGTT_UPDATE ) \
-	param(bool, enable_gvt, false)
+	param(bool, enable_gvt, false) \
+	param(bool, enable_conformance_check, true)
 
 #define MEMBER(T, member, ...) T member;
 struct i915_params {
-- 
2.7.4

